File: http-process/src/main/java/com/github/dragonhht/http/utils/ParameterMap.java
package com.github.dragonhht.http.utils;

import java.util.HashMap;
import java.util.Map;

/**
 * 用于存放请求的参数信息.
 *
 * @author: huang
 * Date: 2018/4/23
 */
public final class ParameterMap extends HashMap {

    private boolean locked = false;

    public ParameterMap() {
        super();
    }

    public ParameterMap(int initialCapacity) {
        super(initialCapacity);
    }

    public ParameterMap(int initialCapacity, float loadFactor) {
        super(initialCapacity, loadFactor);
    }

    public ParameterMap(Map map) {
        super(map);
    }

    public boolean isLocked() {
        return locked;
    }

    public void setLocked(boolean locked) {
        this.locked = locked;
    }

    public void clear() {
        if (locked) throw new IllegalStateException();
        super.clear();
    }

    public Object put(Object key, Object value) {
        if (locked) throw new IllegalStateException();
        return super.put(key, value);
    }

    public void putAll(Map map) {
        if (locked) throw new IllegalStateException();
        super.putAll(map);
    }

    public Object remove(Object key) {
        if (locked) throw new IllegalStateException();
        return super.remove(key);
    }
}

File: netty-server/src/main/java/com/github/dragonhht/http/servlet/ParameterMap.kt
package com.github.dragonhht.http.servlet

import java.util.HashMap

/**
 * .
 *
 * @author: huang
 * @Date: 2019-4-17
 */
internal class ParameterMap<T, R> : HashMap<T, R?> {

    var isLocked = false

    constructor() : super() {}

    constructor(initialCapacity: Int) : super(initialCapacity) {}

    constructor(initialCapacity: Int, loadFactor: Float) : super(initialCapacity, loadFactor) {}

    constructor(map: Map<T, R?>) : super(map) {}

    override fun clear() {
        if (isLocked) throw IllegalStateException()
        super.clear()
    }

    override fun put(key: T, value: R?): R? {
        if (isLocked) throw IllegalStateException()
        return super.put(key, value)
    }

    override fun putAll(map: Map<out T, R?>) {
        if (isLocked) throw IllegalStateException()
        super.putAll(map)
    }

    override fun remove(key: T): R? {
        if (isLocked) throw IllegalStateException()
        return super.remove(key)
    }
}